pkg/api/controller.go

```diff
@@ -802,27 +810,76 @@ func (c *Controller) ListUsers(w http.ResponseWriter, r *http.Request, params Li
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) generateResetPasswordToken(email string) (*string, error) {
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	token, err := GenerateJWTResetPassword(secret, email, currentTime, currentTime.Add(DefaultResetPasswordExpiration))
+	if err != nil {
+		return nil, err
+	}
+	return &token, nil
+}
+
+func (c *Controller) inviteUserRequest(email string) error {
+	// TODO (@shimi9276) create template for sending the email with link for invite using the passed template
+	token, err := c.generateResetPasswordToken(email)
+	if err != nil {
+		return err
+	}
+	err = c.Emailer.SendEmailWithLimit([]string{email}, *token, *token, nil)
+	if err != nil {
+		return err
+	} else {
+		c.Logger.WithField("email", email).Info("invite email sent")
+	}
+	return nil
+}
+
 func (c *Controller) CreateUser(w http.ResponseWriter, r *http.Request, body CreateUserJSONRequestBody) {
+	invite := swag.BoolValue(body.InviteUser)
+	id := body.Id
+	var addr *mail.Address
+	var err error
+	if invite {
+		addr, err = mail.ParseAddress(body.Id)
+		if handleAPIError(w, err) {
+			c.Logger.WithError(err).WithField("email", body.Id).Warn("failed parsing email")
+			return
+		}
+		id = strings.ToLower(addr.Address)
+	}
 	if !c.authorize(w, r, permissions.Node{
 		Permission: permissions.Permission{
 			Action:   permissions.CreateUserAction,
-			Resource: permissions.UserArn(body.Id),
+			Resource: permissions.UserArn(id),
 		},
 	}) {
 		return
 	}
 	u := &model.User{
 		CreatedAt:    time.Now().UTC(),
-		Username:     body.Id,
+		Username:     id,
 		FriendlyName: nil,
 		Source:       "internal",
 	}
+	if invite {
+		u.Email = &addr.Address
+	}
 	ctx := r.Context()
 	c.LogAction(ctx, "create_user")
-	_, err := c.Auth.CreateUser(ctx, u)
+	_, err = c.Auth.CreateUser(ctx, u)
+
 	if handleAPIError(w, err) {
 		return
 	}
+
+	if invite {
+		err = c.inviteUserRequest(id)
```

Comment:
Use the email/addr here and not the id we created for the user. You can also get the created user and pass the email created.

Reply:
Done, used the email from user struct